crates/store/re_types/src/archetypes/asset_video_ext.rs

```diff
@@ -32,30 +32,52 @@ impl AssetVideo {
     #[inline]
     pub fn from_file_contents(contents: Vec<u8>, media_type: Option<impl Into<MediaType>>) -> Self {
         let media_type = media_type.map(Into::into);
-        let media_type = MediaType::or_guess_from_data(media_type, &contents);
-        Self {
-            blob: contents.into(),
-            media_type,
+        if let Some(media_type) = MediaType::or_guess_from_data(media_type, &contents) {
+            Self::new(contents).with_media_type(media_type)
+        } else {
+            Self::new(contents)
         }
     }
 
     /// Determines the presentation timestamps of all frames inside the video.
     ///
     /// Returned timestamps are in nanoseconds since start and are guaranteed to be monotonically increasing.
+    ///
+    /// Panics if the serialized blob data doesn't have the right datatype.
     #[cfg(feature = "video")]
     pub fn read_frame_timestamps_ns(&self) -> Result<Vec<i64>, re_video::VideoLoadError> {
+        use re_types_core::Loggable;
+
         re_tracing::profile_function!();
 
+        let Some(blob) = self.blob.as_ref() else {
+            return Ok(Vec::new());
+        };
+
+        // Grab blob data without a copy.
+        let blob_list_array = blob
+            .array
+            .as_any()
+            .downcast_ref::<arrow::array::ListArray>()
+            .expect("Video blob data is not a ListArray");
```

Comment:
Bit unfortunate to risk a panic in a method that already returns a result anyhow?

Reply:
agreed, but the problem is that `re_video::VideoLoadError` ofc knows nothing about any kind of arrow shenanigans we might be dealing with, which means I'd need to change the error type here or leak things into video load error

Reply:
You can always have an escape hatch like:
```rust
    #[error(transparent)]
    Other(#[from] anyhow::Error),
```
:man_shrugging: